packages/server/lib/crons/trial.ts

```diff
@@ -0,0 +1,111 @@
+import {
+    AnalyticsTypes,
+    analytics,
+    disableScriptConfig,
+    errorNotificationService,
+    getAccountWithFinishedTrialAndSyncs,
+    getTrialsApproachingExpiration,
+    syncManager,
+    updatePlan,
+    userService
+} from '@nangohq/shared';
+import { flagHasPlan, getLogger, metrics, report } from '@nangohq/utils';
+
+import { sendTrialAlmostOverEmail } from '../helpers/email.js';
+import { getOrchestrator } from '../utils/utils.js';
+
+import type { Lock } from '@nangohq/kvstore';
+
+const cronMinutes = 120;
+const daysBeforeTrialIsOver = 3;
+
+const logger = getLogger('cron.trial');
+
+export function trialCron(): void {
+    if (!flagHasPlan) {
+        return;
+    }
+
+    cron.schedule(
+        `*/${cronMinutes} * * * *`,
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
+        async () => {
+            const start = Date.now();
+            try {
+                await tracer.trace<Promise<void>>('nango.cron.trial', async () => {
+                    await exec();
+                });
+
+                logger.info('✅ done');
+            } catch (err) {
+                report(new Error('cron_failed_to_check_trial', { cause: err }));
+            }
+            metrics.duration(metrics.Types.CRON_TRIAL, Date.now() - start);
+        }
+    );
+}
+
+export async function exec(): Promise<void> {
+    const locking = await getLocking();
+
+    logger.info(`Starting`);
+
+    const lockKey = `lock:trial:cron`;
+    let lock: Lock | undefined;
+    try {
+        try {
+            lock = await locking.acquire(lockKey, 60 * 1000);
+        } catch (err) {
+            logger.info(`Could not acquire lock, skipping`, err);
+            return;
+        }
+
+        // Send email to team that are close to expiration
+        const res = await getTrialsApproachingExpiration(db.knex, { daysLeft: daysBeforeTrialIsOver });
+        if (res.isOk()) {
+            for (const plan of res.value) {
+                await updatePlan(db.knex, { id: plan.id, trial_end_notified_at: new Date() });
+
+                logger.info('Trial soon to be over for account', plan.account_id);
+                void analytics.track(AnalyticsTypes.ACCOUNT_TRIAL_EXPIRING_MAIL, plan.account_id);
+
+                const users = await userService.getUsersByAccountId(plan.account_id);
+
+                // Send in parallel
+                await Promise.all(
+                    users.map(async (user) => {
+                        if (!user.email_verified) {
+                            return;
+                        }
+
+                        logger.info('  Sending mail to', user.id);
+                        await sendTrialAlmostOverEmail({ user, inDays: daysBeforeTrialIsOver });
+                    })
+                );
+            }
+        }
+
+        // Disable all scripts
+        const orchestrator = getOrchestrator();
+        const accountsToPause = await getAccountWithFinishedTrialAndSyncs(db.knex);
```

Comment:
This query is getting all trial over in the last 24 hours, it's a bit dumb because I don't know which one I have paused already. I probably need to store a boolean in the plan (same problem with the email at the end of the trial that will need to be send only once :D)
Happy to get suggestion

Reply:
it is a bit dumb indeed and maybe one day we will need way to disable everything in bulk but for now I feel it would be ok.
I agree however about storing a boolean or even maybe a `trial_disabled_at` date to not re-process the same plan again